vespa-cloud/album-recommendation-docproc/src/main/java/ai/vespa/example/album/LyricsDocumentProcessor.java

```diff
@@ -76,22 +72,23 @@ public Progress process(Processing processing) {
                 DocumentPut put = (DocumentPut) op;
                 Document document = put.getDocument();
                 if (document.getDataType().isA(MUSIC_DOCUMENT_TYPE)) {
-
-                    // Set lyrics if already looked up, and set DONE
-                    FieldValue lyrics = (FieldValue)processing.getVariable(VAR_LYRICS);
-                    if (lyrics != null) {
-                        document.setFieldValue("lyrics", lyrics);
-                        logger.info("  Set lyrics, Progress.DONE");
-                        return DocumentProcessor.Progress.DONE;
-                    }
-
-                    // Make a lyric request if not already pending
-                    if (processing.getVariable(VAR_REQ_ID) == null) {
+                    if (processing.getVariable(VAR_REQ_ID) != null) { // A request has been made, check for response
+                        long reqId = (long)processing.getVariable(VAR_REQ_ID);
+                        Document doc = responses.get(reqId);
+                        if (doc != null) {
```

Comment:
Here you look at the response and do the complete error handling here.

Reply:
right. refactored as well so the code should be also easier to read now